docs/API.md

```diff
@@ -956,7 +963,11 @@ try {
 `public void putObject(String bucketName, String objectName, InputStream stream, long size, String contentType,
 		  SecretKey key)`
 
-Encrypts and then uploads an object from an InputStream.
+Takes data from given stream, encrypts it using a random content key and uploads it as object to given bucket. Also
+uploads the encrypted content key and iv as header of the encrypted object. The content key is encrypted using the
+master key passed to this function.
+
+If the object is larger than 5MB, the client will automatically use a multipart session.
```

Comment:
All objects are treated as unknown size case because of encryption ie if object size is given as 13MB, it will be uploaded as single putobject because for unknown sized object, multipart works > 525MiB.